rust/src/mqtt/mqtt.rs

```diff
@@ -589,6 +608,17 @@ impl MQTTState {
         tx.tx_data.set_event(event as u8);
         self.transactions.push_back(tx);
     }
+
+fn mqtt_frames(&mut self,flow: *const Flow, stream_slice: &StreamSlice, input: &MQTTMessage)  {
+        let hdr = stream_slice.as_slice();
+            //MQTT payload has a fixed header of 2 bytes
+        let _mqtt_ts_hdr = Frame::new(flow, stream_slice, hdr,2 , MQTTFrameType::Header as u8);
+            SCLogDebug!("nfs4_ts_hdr Frame {:?}", _mqtt_ts_hdr);
+        let data= &hdr[1..input.header.remaining_length as usize];
+        let _mqtt_ts_data = Frame::new(flow, stream_slice, data, input.header.remaining_length as i64, MQTTFrameType::Data as u8);
+            SCLogDebug!("nfs4_ts_ops Frame {:?}", _mqtt_ts_data);
```

Comment:
This is still a WIP, I just wanted to get feedback if I am going in the right direction or not?